homeassistant/components/fritz/config_flow.py

```diff
@@ -0,0 +1,244 @@
+"""Config flow to configure the FRITZ!Box Tools integration."""
+import logging
+from urllib.parse import urlparse
+
+from fritzconnection.core.exceptions import FritzConnectionException, FritzSecurityError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.ssdp import (
+    ATTR_SSDP_LOCATION,
+    ATTR_UPNP_FRIENDLY_NAME,
+    ATTR_UPNP_UDN,
+)
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_PORT, CONF_USERNAME
+from homeassistant.core import callback
+
+from .common import FritzBoxTools
+from .const import (
+    DEFAULT_HOST,
+    DEFAULT_PORT,
+    DOMAIN,
+    ERROR_AUTH_INVALID,
+    ERROR_CONNECTION_ERROR,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class FritzBoxToolsFlowHandler(ConfigFlow):
+    """Handle a FRITZ!Box Tools config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize FRITZ!Box Tools flow."""
+        self._host = None
+        self._entry = None
+        self._name = None
+        self._password = None
+        self._port = None
+        self._username = None
+        self.import_schema = None
+        self.fritz_tools = None
+
+    async def fritz_tools_init(self):
+        """Initialize FRITZ!Box Tools class."""
+        self.fritz_tools = FritzBoxTools(
+            hass=self.hass,
+            host=self._host,
+            port=self._port,
+            username=self._username,
+            password=self._password,
+        )
+
+        try:
+            await self.fritz_tools.async_setup()
+        except FritzSecurityError:
+            return ERROR_AUTH_INVALID
+        except FritzConnectionException:
+            return ERROR_CONNECTION_ERROR
+
+        return None
+
+    @callback
+    def _async_create_entry(self):
+        """Async create flow handler entry."""
+        return self.async_create_entry(
+            title=self._name,
+            data={
+                CONF_HOST: self.fritz_tools.host,
+                CONF_PASSWORD: self.fritz_tools.password,
+                CONF_PORT: self.fritz_tools.port,
+                CONF_USERNAME: self.fritz_tools.username,
+            },
+        )
+
```

Comment:
We need to implement `async_step_import` here to handle the legacy config being imported from the device_tracker platform setup

Reply:
Done